borg/remote.py

```diff
@@ -164,11 +167,21 @@ def __init__(self, location, create=False, lock_wait=None, lock=True, args=None)
         self.id = self.call('open', location.path, create, lock_wait, lock)
 
     def __del__(self):
-        self.close()
+        if self.p:
+            self.close()
+            assert False, "cleanup happened in Repository.__del__"
```

Comment:
For me all appearances of `__del__` cleanup are gone in the testsuite except the one in test_remote_repo_restrict_to_path (which I think is caused by `patch.object` basically overriding the context manager implemented in Repository, i.e. the patched object is bound instead of the Repository).